tap_universal_file/streams.py

```diff
@@ -478,3 +479,234 @@ def _get_readers(
                     file_name,
                     file["last_modified"],
                 )
+
+
+class ParquetStream(FileStream):
+    """Stream for reading Parquet files."""
+
+    def get_rows(self) -> Generator[dict[str, Any], None, None]:
+        """Retrive all rows from all Parquet files.
+
+        Yields:
+            A dictionary containing information about a row in a Parquet file.
+        """
+        for reader, file_name, last_modified in self._get_readers():
+            self.logger.info("Starting sync of %s.", file_name)
+            line_number = 0
+            for row in reader.to_pylist():
+                line_number += 1
+                yield self.add_additional_info(
+                    row=self._pre_process(row),
+                    file_name=file_name,
+                    line_number=line_number,
+                    last_modified=last_modified,
+                )
+            self.logger.info(
+                "Completed sync of %s records from %s.",
+                line_number,
+                file_name,
+            )
+
+    def get_properties(self) -> dict:
+        """Get a list of properties for an Avro file, to be used in creating a schema.
+
+        Returns:
+            A list of properties representing an Avro file.
+        """
+        properties = {}
+        for field in self._get_fields():
+            properties.update(self._get_property(field))
+        return properties
+
+    def _get_fields(self) -> Generator[dict | str, None, None]:
+        """Gets all fields in an avro file from schema and configured coercion strategy.
+
+        Raises:
+            ValueError: If the provided coercion strategy is invalid.
+
+        Yields:
+            A dictionary or string representing a field.
+        """
+        strategy = self.config["parquet_type_coercion_strategy"]
+        if strategy == "convert":
+            for reader, _, _ in self._get_readers():
```

Comment:
I find these few lines really hard to read. 

Why not include the names of the three variables `self._get_readers()` returns even if we don't use them it is a lot more clear that we're just reading the files that match with our file path here.

Reply:
Violation of Ruff B007, but I can understand that it's more readable with the variable names included. Changed to give them all names.

Reply:
Sure, can you add a comment then explaining what this function is returning

Reply:
Reverted to underscores but added an explanation in comments.